### PR TITLE
rebar3: skip failing tests on M1 chip

### PR DESCRIPTION
**Pull request by jaen**
###### Motivation for this change

There are a few rebar3 tests that fail on an M1 Mac. This PR disables them, so that rebar3 can at least be installed

See https://github.com/NixOS/nixpkgs/issues/130012 for details.

###### Things done

I have run `nixpkgs-review` on a NixOS machine, but on an M1 Mac it failed, not sure how much this matters.
I have also run build on the M1 and it worked (duh), but Nix apparently doesn't have sandbox enabled on that machine for whatever reason (didn't change any defaults), so I didn't mark the checkbox.

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
@dlesl I'm not sure which machine you are on most of the time.
A little summary of the issue is that some tests fail on an M1. 
Those tests failure appear to be related to file permissions.
I don't know much about the issue itself, I'm just tagging you in case you have more information.
I realise you might be busy, no worries at all if you can't reply.

I'm planning to merge this next monday unless there is a manifestation from someone.

@jaen just out of curiosity. Did you try to build rebar3 from source on an M1? (suggested by Thomas, it could be a way to find out if it is an M1 problem or not).

**Comment by jaen** [comment]
@happysalada as I mentioned in the other thread, I've tried it just now in a `nix shell nixpkgs#beam.packages.erlangR23.erlang` and seems to fail the same way.

**Comment by happysalada** [comment]
If possible, can you file a bug upstream and add a comment on top of the patch that references the bug. Just so we know when we can remove the patch.

@DianaOlympos well spotted!

File: pkgs/development/tools/build-managers/rebar3/default.nix [hunk 1]
```diff
@@ -39,6 +39,8 @@ let
       HOME=. escript bootstrap
     '';
 
+    patches = if (stdenv.isDarwin && stdenv.isAarch64) then [ ./m1-tests-skip.patch ] else [];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/rebar3/default.nix:42]:
```suggestion
    patches = lib.optionals (stdenv.isDarwin && stdenv.isAarch64) [ ./m1-tests-skip.patch ];
```

Comment (jaen) [reply pkgs/development/tools/build-managers/rebar3/default.nix:42]: I can, but I don't quite understand how this differs if it's still an if underneath? Does it somehow affect what ends up in the store and using `lib.optionals` makes sure that someone won't lose the file even if they're not on an M1?

Comment (happysalada) [reply pkgs/development/tools/build-managers/rebar3/default.nix:42]: it is a conditional, yes.
It's a little more idiomatic in nixpkgs, it absolutely does not change the logic with what you had.

Comment (jaen) [reply pkgs/development/tools/build-managers/rebar3/default.nix]: Used optionals and added a comment about removing then on next release

**Comment by happysalada** [comment]
@jaen Thank you for this!

**Pull request merged**